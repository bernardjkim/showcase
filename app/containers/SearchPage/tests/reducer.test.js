import { fromJS } from 'immutable';
import searchPageReducer from '../reducer';

import {
  loadArticles,
  loadArticlesSuccess,
  loadArticlesError,
} from '../actions';

describe('searchPageReducer', () => {
  let state;
  beforeEach(() => {
    state = fromJS({
      loading: false,
      error: false,
      articles: false,
    });
  });

  it('returns the initial state', () => {
    expect(searchPageReducer(state, {})).toMatchSnapshot();
  });

  it('handle the load articles action', () => {
    const query = { q: 'q' };
    expect(searchPageReducer(state, loadArticles(query))).toMatchSnapshot();
  });

  it('handle the load articles success action', () => {
    const articles = [];
    expect(
      searchPageReducer(state, loadArticlesSuccess(articles)),
    ).toMatchSnapshot();
  });

  it('handle the load articles error action', () => {
    const error = new Error('Test Error');
    expect(
      searchPageReducer(state, loadArticlesError(error)),
    ).toMatchSnapshot();
  });
});
