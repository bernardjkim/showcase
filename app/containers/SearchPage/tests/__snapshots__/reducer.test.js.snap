// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchPageReducer handle the clear state action 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "articles": Immutable.List [],
  "search": "",
  "offset": 0,
}
`;

exports[`searchPageReducer handle the load articles action 1`] = `
Immutable.Map {
  "loading": true,
  "error": false,
  "articles": Immutable.List [],
  "search": "",
  "offset": 0,
}
`;

exports[`searchPageReducer handle the load articles error action 1`] = `
Immutable.Map {
  "loading": false,
  "error": [Error: Test Error],
  "articles": Immutable.List [],
  "search": "",
  "offset": 0,
}
`;

exports[`searchPageReducer handle the load articles error action 2`] = `
Immutable.Map {
  "loading": false,
  "error": [Error: Test Error],
  "articles": Immutable.List [],
  "search": "",
  "offset": 0,
}
`;

exports[`searchPageReducer handle the load articles success action 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "articles": Immutable.List [],
  "search": "",
  "offset": 0,
}
`;

exports[`searchPageReducer handle the load next action 1`] = `
Immutable.Map {
  "loading": true,
  "error": false,
  "articles": Immutable.List [],
  "search": "",
  "offset": 0,
}
`;

exports[`searchPageReducer handle the load next success action 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "articles": Immutable.List [],
  "search": "",
  "offset": 0,
}
`;

exports[`searchPageReducer handle the set search action 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "articles": Immutable.List [],
  "search": "search",
  "offset": 0,
}
`;

exports[`searchPageReducer returns the initial state 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "articles": Immutable.List [],
  "search": "",
  "offset": 0,
}
`;
